backend/app/usecase/search/search.go

```diff
@@ -66,13 +71,85 @@ func (s Search) searchResource(resource Resource, orderBy order.Order, query Que
 }
 
 func (s Search) searchShortLink(query Query, orderBy order.Order, filter Filter) (Result, error) {
-	return Result{}, nil
+	if query.User == nil {
+		s.logger.Error(errors.New("user not provided"))
+		return Result{}, nil
+	}
+
+	shortLinks, err := s.getShortLinkByUser(*query.User)
+	if err != nil {
+		return Result{}, err
+	}
+
+	var matchedAliasByAll, matchedAliasByAny, matchedLongLinkByAll, matchedLongLinkByAny []entity.ShortLink
+	keywords := getKeywords(query.Query, " ")
+	for _, shortLink := range shortLinks {
+		if matcher.ContainsAll(keywords, shortLink.Alias) {
+			matchedAliasByAll = append(matchedAliasByAll, shortLink)
+		} else if matcher.ContainsAny(keywords, shortLink.Alias) {
+			matchedAliasByAny = append(matchedAliasByAny, shortLink)
+		} else if matcher.ContainsAll(keywords, shortLink.LongLink) {
+			matchedLongLinkByAll = append(matchedLongLinkByAll, shortLink)
+		} else if matcher.ContainsAny(keywords, shortLink.LongLink) {
+			matchedLongLinkByAny = append(matchedLongLinkByAny, shortLink)
+		}
+	}
```

Comment:
Let's break this into 4 separate loops for now. We can look into simplification in the future.

Reply:
Discussed on Slack. Postpone this to the future.